apps/google-analytics-4/frontend/src/locations/Sidebar.tsx

```diff
@@ -1,7 +1,21 @@
 import AnalyticsApp from 'components/main-app/analytics-app/AnalyticsApp';
+import { useSDK } from '@contentful/react-apps-toolkit';
+
+// TO DO: Add propertyId and reportSlug logic
 
 const Sidebar = () => {
-  return <AnalyticsApp />;
+  const sdk = useSDK<any>();
+
+  const { serviceAccountKey, serviceAccountKeyId } = sdk.parameters;
```

Comment:
Now that we've cleaned this up it leads me to think we _could_ -- instead of passing these low level keys which are only used to set up the API -- just initialize the api here (via useApi) and pass it down directly.

But I'd save for a later refactor if you agree 😄

Reply:
oh I do like that idea very much, a good separation of concerns. 👍